micro-rdk/src/common/ota.rs

```diff
@@ -159,11 +158,30 @@ impl OtaService {
                 )))
             }
         };
+
+        #[cfg(not(feature = "esp32"))]
+        let max_size = 1024 * 1024 * 4; // 4MB
+        #[cfg(feature = "esp32")]
+        let max_size = {
+            log::debug!("getting size of dst partition...");
+            let ptr: *const esp_partition_t =
+                unsafe { esp_ota_get_next_update_partition(std::ptr::null()) };
+            if ptr.is_null() {
+                return Err(OtaError::Other(
+                    "failed to get dst partition info - may not exist".to_string(),
+                ));
+            }
+            let size = unsafe { (*ptr).size };
```

Comment:
this field is a u32 according to esp-idf-svc [docs](https://docs.esp-rs.org/esp-idf-svc/esp_idf_sys/struct.esp_partition_t.html)